src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -145,6 +149,26 @@ private AnalyzerConfigSet(ImmutableArray<AnalyzerConfig> analyzerConfigs, Global
             Debug.Assert(allMatchers.Count == _analyzerConfigs.Length);
 
             _analyzerMatchers = allMatchers.ToImmutableAndFree();
+
+        }
+
+        /// <summary>
+        /// Gets an <see cref="AnalyzerConfigOptionsResult"/> that contain the options that apply globally
+        /// </summary>
+        public AnalyzerConfigOptionsResult GlobalConfigOptions
+        {
+            get
+            {
+                if (_lazyConfigOptions is null)
+                {
+                    Interlocked.CompareExchange(
```

Comment:
What is the value in using this lazy pattern given that `AnalyzerConfigOptionsResult` is a `struct`? This property will always return a copy of the value, the `ParseGlobalConfigOptions` can run multiple times even in this config. Can't see the value `StrongBox<AnalyzerConfigOptionResult>` provides over `AnalyzerConfigOptionResult?` (nullable value). 

What am I missing here?

Reply:
Because the struct is bigger than an IntPtr, assignment isn't guaranteed to be atomic (AFAIK). Means one thread could read from it and get a half initialized struct if we don't box + interlock exchange.

---
In reply to: [475294138](https://github.com/dotnet/roslyn/pull/45871#discussion_r475294138) [](ancestors = 475294138)